### DISCUSSION TITLE
Changing Icons

### DISCUSSION THREAD

**Discussion by Ethanol48** [NvChad/NvChad#2007] [Q&A] [answered]
I'm trying to changing the icon for Solidity files, I'm doing this in my override.lua:

```lua 


M.ui = {

  icons = {
    devicons = {

      default = {
        icon = ""
      },

      solidity = {
        icon = "",
      },
    },

    solidity = {
      icon = "",
    },

  },

```

I put the overrides under icons.devicons and icons and they don't work

**Comment by Ethanol48** [comment]
i put the wrong name in default, is default_icon, I changed it and it doesn't work either

**Reply by siduck** [reply, marked as answer]
```lua
{
    "nvim-tree/nvim-web-devicons",
    opts = {
      override = {
        lua = {
          icon = "a",
          name = "lua",
        },
        c = {
          icon = "c",
          name = "lua",
        },
      },
    },
  }
  ```
  
  
![image](https://github.com/NvChad/NvChad/assets/59060246/ef9d7d07-02de-4441-9fd2-cbdbb0d45f5b)


M.ui wont work, because all options in chadrc are meant to change the options present in core/default_config.lua so they must  maintain that table structure.

As you are trying to override filetype icons, override "opts" spec ( check lazy.nvim docs to know whats a spec ) of the webdevicons plugin
